packages/aws_common/lib/src/http/aws_http_client_io.dart

```diff
@@ -441,7 +443,7 @@ class AWSHttpClientImpl extends AWSHttpClient {
   }) {
     final requestProgressController = StreamController<int>.broadcast();
     final responseProgressController = StreamController<int>.broadcast();
-    final cancelTrigger = Completer<void>();
+    final cancelTrigger = Completer<void>.sync();
```

Comment:
nit: Could we have a little comment explaining why it's safe to use `sync` constructor here?

Reply:
Done 👍